include/experimental/__p0009_bits/layout_stride.hpp

```diff
@@ -199,7 +199,15 @@ struct layout_stride {
       template<class IntegralType>
       MDSPAN_INLINE_FUNCTION
       static constexpr const __strides_storage_t fill_strides(const std::array<IntegralType,extents_type::rank()>& s) {
+        // avoid warning for use of host std::array operator[]
+  #if defined(_MDSPAN_HAS_CUDA) || defined(_MDSPAN_HAS_HIP)
+        const IntegralType* s_ptr = reinterpret_cast<const IntegralType*>(&s);
```

Comment:
We need to figure out what to do properly here, since `reinterpret_cast` can't be used in a constant expression, so we will remove it for now

Reply:
`std::array` isn't usable on Cuda devices anyway unless you are using `-expt-relaxed-constexpr`, right? Wouldn't an interface in terms of `C-style` arrays be a more natural choice?

Reply:
Yes and no: we have constructors which have to take std::array because that is the standard. Also a bunch of operations actually do work on device (i.e. construction, destruction, structured binding - because they don't go through member functions, or at least go through defaulted member functions).

That said the above casting approach doesn't work in constexpr so we changed this.